lib/internal/Magento/Framework/DB/AbstractMapper.php

```diff
@@ -124,7 +124,7 @@ public function map(CriteriaInterface $criteria)
             $mapperMethod = 'map' . $camelCaseKey;
             if (method_exists($this, $mapperMethod)) {
                 if (!is_array($value)) {
-                    $value = [$value];
+                    throw \Exception('Wrong type of argument, expecting array for '. $mapperMethod);
```

Comment:
`new` operator is missing here. Looks like `InvalidArgumentException` is more suitable.

Besides fixing an existing unit test
```
There was 1 error:
1) Magento\Framework\DB\Test\Unit\AbstractMapperTest::testMap with data set #0 (array('mapMyMapperMethodOne', 'mapMyMapperMethodTwo'), array('my-test-value1', 'my-test-value2'))
Error: Call to undefined function Exception()
/home/travis/build/magento/magento2/lib/internal/Magento/Framework/DB/AbstractMapper.php:127
/home/travis/build/magento/magento2/lib/internal/Magento/Framework/DB/Test/Unit/AbstractMapperTest.php:141
```
please cover this new logic with additional test method.

Reply:
Thanks. Will update test also in next commit.

Reply:
Thanks. Will update test also in next commit.